coderd/apikey.go

```diff
@@ -67,18 +67,39 @@ func (api *API) postToken(rw http.ResponseWriter, r *http.Request) {
 	}
 
 	// Map and validate requested scope.
-	// Accept special scopes (all, application_connect) and curated public low-level scopes.
-	scopes := database.APIKeyScopes{database.APIKeyScopeAll}
-	if createToken.Scope != "" {
+	// Accept legacy special scopes (all, application_connect) and external scopes.
+	// Default to coder:all scopes for backward compatibility.
+	scopes := database.APIKeyScopes{database.ApiKeyScopeCoderAll}
+	if len(createToken.Scopes) > 0 {
+		scopes = make(database.APIKeyScopes, 0, len(createToken.Scopes))
+		for _, s := range createToken.Scopes {
+			name := string(s)
+			if !rbac.IsExternalScope(rbac.ScopeName(name)) {
+				httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+					Message: "Failed to create API key.",
+					Detail:  fmt.Sprintf("invalid or unsupported API key scope: %q", name),
+				})
+				return
+			}
+			scopes = append(scopes, database.APIKeyScope(name))
```

Comment:
Do we need to convert `all` & `application_connect` here?

Reply:
I don't think so. 

Say someone uses the (new) scopes slices to submit multiple scopes, I'd expect them to use the new scope names instead of the legacy ones. So I'd rather return an error here, and only add the compat for the "old" singular Scope field.